File: Backend/ConstructionProjectTracker.API/Entities/TaskProgressLog.cs
namespace ConstructionProjectTracker.API.Entities;

public class TaskProgressLog
{
    public int Id { get; set; }
    public int TaskId { get; set; }
    public int EngineerId { get; set; }
    public int PreviousProgress { get; set; }
    public int NewProgress { get; set; }
    public string Description { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public TaskItem Task { get; set; } = null!;
    public Engineer Engineer { get; set; } = null!;
}

File: Frontend/construction-project-tracker/src/app/core/models/task-progress-log.ts
export interface TaskProgressLog {
  id: number;
  taskId: number;
  engineerId: number;
  engineerName: string;
  previousProgress: number;
  newProgress: number;
  description: string;
  createdAt: string;
}

export interface CreateTaskProgressLog {
  newProgress: number;
  description: string;
}
